src/mlpack/core/optimizers/sgd/sgd.hpp

```diff
@@ -135,6 +136,9 @@ class SGD
   //! Modify whether or not the individual functions are shuffled.
   bool& Shuffle() { return shuffle; }
 
+  //! Get the update policy. 
+  UpdatePolicy GetUpdatePolicy() const { return updatePolicy; }
```

Comment:
Instead of using ``GetUpdatePolicy`` because of the template parameter shadowed issue, you can just rename the template parameter to ``updatePolicyType`` that way we are still consistent with the rest of the codebase.